libs/execution_context/ebpf_maps.c

```diff
@@ -425,22 +523,102 @@ _update_hash_map_entry(_In_ ebpf_core_map_t* map, _In_ const uint8_t* key, _In_o
         (ebpf_hash_table_find((ebpf_hash_table_t*)map->data, key, &value) != EBPF_SUCCESS))
         result = EBPF_INVALID_ARGUMENT;
     else
-        result = ebpf_hash_table_update((ebpf_hash_table_t*)map->data, key, data);
+        result = ebpf_hash_table_update((ebpf_hash_table_t*)map->data, key, data, NULL);
 
     return result;
 }
 
 static ebpf_result_t
-_delete_hash_map_entry(_In_ ebpf_core_map_t* map, _In_ const uint8_t* key)
+_update_hash_map_entry_with_handle(
+    _In_ ebpf_core_map_t* map,
+    _In_ const uint8_t* key,
+    ebpf_object_type_t value_type,
+    _In_ const uint8_t* new_value,
+    uintptr_t value_handle)
+{
+    ebpf_result_t result = EBPF_SUCCESS;
+    ebpf_lock_state_t lock_state;
+    size_t entry_count = 0;
+    if (!map || !key || !new_value)
+        return EBPF_INVALID_ARGUMENT;
+
+    // Convert value handle to an object pointer.
+    struct _ebpf_object* object;
+    int return_value = ebpf_reference_object_by_handle(value_handle, value_type, &object);
+    if (return_value != EBPF_SUCCESS)
+        return return_value;
+
+    // Validate that the object's program type is
+    // not in conflict with the map's program type.
+    const ebpf_program_type_t* program_type = (object->get_program_type) ? object->get_program_type(object) : NULL;
```

Comment:
Does get_program_type apply to maps or link objects? If not why add it to the base ebpf_object_t?

Reply:
yes it definitely applies to maps.   (I don't know whether we will need it to apply to links as well, but it would make sense)

Reply:
Updated comment in ebpf_object.h